File: CS/Modules.Chart/ChartModule.cs
using Prism.Modularity;
using Prism.Regions;

namespace Modules.Chart {
    public class ChartModule: IModule {
        private readonly IRegionManager regionManager;
        public ChartModule(IRegionManager regionManager) {
            this.regionManager = regionManager;
        }
        void IModule.Initialize() {
            regionManager.RegisterViewWithRegion("ChartRegion", typeof(ChartView));
        }
    }
}

File: VB/Modules.Chart/ChartModule.vb
Imports Prism.Regions
Imports Prism.Modularity

Namespace Modules.Chart
    Public Class ChartModule
        Implements IModule
        Private ReadOnly regionManager As IRegionManager
        Public Sub New(ByVal regionManager As IRegionManager)
            Me.regionManager = regionManager
        End Sub
        Public Sub Initialize() Implements IModule.Initialize
            regionManager.RegisterViewWithRegion("ChartRegion", GetType(ChartView))
        End Sub
    End Class
End Namespace
